File: backend/amigos_java_backend/src/main/java/com/socialmedia/entity/BlockedUser.java
package com.socialmedia.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "blocked_users", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"blocker_id", "blocked_user_id"})
})
@Getter
@Setter
public class BlockedUser {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "blocker_id", nullable = false)
    private User blocker;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "blocked_user_id", nullable = false)
    private User blockedUser;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;
}

File: backend/amigos_dotnet_backend/Models/Entities/BlockedUser.cs
using SocialMediaAdminBackend.Models.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SocialMediaAdminBackend.Models.Entities;

[Table("blocked_users")]
public class BlockedUser
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("blocker_id")]
    public long BlockerId { get; set; }

    [Column("blocked_user_id")]
    public long BlockedUserId { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [ForeignKey("BlockerId")]
    public User? Blocker { get; set; }

    [ForeignKey("BlockedUserId")]
    public User? BlockedUserEntity { get; set; }
}